packages/jest-haste-map/src/index.js

```diff
@@ -189,6 +190,7 @@ class HasteMap {
       roots: options.roots,
       useWatchman:
         options.useWatchman == null ? true : options.useWatchman,
+      throwOnModuleCollision: options.throwOnModuleCollision,
```

Comment:
Because this option is optional, we should do `!!options.throwOnModuleCollision`